File: app/src/main/java/com/vsa/filmoteca/data/repository/sharedpreferences/SharedPreferencesManager.java
package com.vsa.filmoteca.data.repository.sharedpreferences;

import android.content.Context;
import android.content.SharedPreferences;

import javax.inject.Inject;

/**
 * Created by seldon on 27/03/15.
 */
public class SharedPreferencesManager {

    //SharedPreferences
    private static final String SHAREDPREFERENCES_NAME = "MySharedPreferences";
    private static final String SHAREDPREFERENCES_CURRENT = "Current";
    private static final String SHAREDPREFERENCES_SIZE = "Size";
    private static final String SHAREDPREFERENCES_TWITTER_PROFILE_IMAGE_URL = "TwitterProfileImageUrl";
    private static final String SHAREDPREFERENCES_TWITTER_USER_DESCRIPTION = "TwitterUserDescription";


    private SharedPreferences mSharedPreferences;

    @Inject
    public SharedPreferencesManager(Context context) {
        mSharedPreferences = context.getSharedPreferences(SHAREDPREFERENCES_NAME, Context.MODE_PRIVATE);
    }

    public int getCurrentMovieIndex() {
        return mSharedPreferences.getInt(SHAREDPREFERENCES_CURRENT, 0);
    }

    public void setCurrentMovieIndex(int index) {
        mSharedPreferences
                .edit()
                .putInt(SHAREDPREFERENCES_CURRENT, index)
                .commit();
    }

    public int getMoviesCount() {
        return mSharedPreferences.getInt(SHAREDPREFERENCES_SIZE, 0);
    }

    public void setMoviesCount(int index) {
        mSharedPreferences
                .edit()
                .putInt(SHAREDPREFERENCES_SIZE, index)
                .commit();
    }

    public String getTwitterProfileImageUrl() {
        return mSharedPreferences.getString(SHAREDPREFERENCES_TWITTER_PROFILE_IMAGE_URL, "");
    }

    public void setTwitterProfileImageUrl(String url) {
        mSharedPreferences
                .edit()
                .putString(SHAREDPREFERENCES_TWITTER_PROFILE_IMAGE_URL, url)
                .commit();
    }

    public String getTwitterUserDescription() {
        return mSharedPreferences.getString(SHAREDPREFERENCES_TWITTER_USER_DESCRIPTION, "");
    }

    public void setTwitterUserDescription(String userDescription) {
        mSharedPreferences
                .edit()
                .putString(SHAREDPREFERENCES_TWITTER_USER_DESCRIPTION, userDescription)
                .commit();
    }

    public void removeTwitterAccountInfo() {
        mSharedPreferences
                .edit()
                .remove(SHAREDPREFERENCES_TWITTER_PROFILE_IMAGE_URL)
                .remove(SHAREDPREFERENCES_TWITTER_USER_DESCRIPTION)
                .commit();
    }


}

File: app/src/main/java/com/vsa/filmoteca/data/repository/sharedpreferences/SharedPreferencesManager.kt
package com.vsa.filmoteca.data.repository.sharedpreferences

import android.content.Context
import android.content.SharedPreferences

import javax.inject.Inject

/**
 * Created by seldon on 27/03/15.
 */
class SharedPreferencesManager @Inject
constructor(context: Context) {

    companion object {
        private const val KEY_NAME = "MySharedPreferences"
        private const val KEY_CURRENT = "Current"
        private const val KEY_SIZE = "Size"
    }

    private val mSharedPreferences: SharedPreferences = context.getSharedPreferences(KEY_NAME, Context.MODE_PRIVATE)

    var currentMovieIndex: Int
        get() = mSharedPreferences.getInt(KEY_CURRENT, 0)
        set(index) {
            mSharedPreferences
                    .edit()
                    .putInt(KEY_CURRENT, index)
                    .apply()
        }

    var moviesCount: Int
        get() = mSharedPreferences.getInt(KEY_SIZE, 0)
        set(index) {
            mSharedPreferences
                    .edit()
                    .putInt(KEY_SIZE, index)
                    .apply()
        }

}
